File: Exchange.Machine.Domain/Coin.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Exchange.Machine.Domain
{
    public class Coin : ICoin
    {
        private readonly byte _value;
        private int _quantity;

        public byte Value { get { return _value; } }
        public int Quantity { get { return _quantity; } }
        public int Total { get { return _value * _quantity; } }
        public bool IsAvailable { get { return _quantity > 0; } }

        public Coin(byte value, int quantity)
        {
            _value = value;
            _quantity = quantity;
        }

        public void IncreaseQuantity(int quantity) => _quantity += quantity;
        public bool DecreaseQuantity(int quantity = 1)
        {
            var isValid = IsValidQuantityDecrease(quantity);

            if (isValid)
                _quantity -= quantity;

            return isValid;
        }

        private bool IsValidQuantityDecrease(int quantity) => _quantity >= quantity;

        public static Coin[] New() => new[]
        {
            new Coin(CoinEnumerator.Coin1.ToByte(), default),
            new Coin(CoinEnumerator.Coin5.ToByte(), default),
            new Coin(CoinEnumerator.Coin10.ToByte(), default),
            new Coin(CoinEnumerator.Coin25.ToByte(), default),
            new Coin(CoinEnumerator.Coin50.ToByte(), default),
            new Coin(CoinEnumerator.Coin100.ToByte(), default)
        };
    }

    public static class CoinExtension
    {
        public static string Formatting(this int coin, double formula = 100) => (coin / formula).ToString("0.00");
    }
}

File: ExchangeMachine/ClientApp/src/app/cash-register/coin/coin.ts
import { CashRegisterConst } from "../cash-register-const";

export class Coin {

    newCoins(coin1, coin5, coin10, coin25, coin50, coin100) {
        let coins = [];

        if (coin1.value)
            coins.push({ value: CashRegisterConst.Coins.C1, quantity: coin1.value });

        if (coin5.value)
            coins.push({ value: CashRegisterConst.Coins.C5, quantity: coin5.value });

        if (coin10.value)
            coins.push({ value: CashRegisterConst.Coins.C10, quantity: coin10.value });

        if (coin25.value)
            coins.push({ value: CashRegisterConst.Coins.C25, quantity: coin25.value });

        if (coin50.value)
            coins.push({ value: CashRegisterConst.Coins.C50, quantity: coin50.value });

        if (coin100.value)
            coins.push({ value: CashRegisterConst.Coins.C100, quantity: coin100.value });

            return coins;
    }
}
  